src/cli/cmd_build_and_check.rs

```diff
@@ -104,22 +102,23 @@ pub fn cmd_check(args: CmdCheck) -> miette::Result<()> {
         .path
         .unwrap_or_else(|| std::env::current_dir().unwrap().try_into().unwrap());
 
+    // TODO: field type should be determined by a flag
     // produce all TASTs and stop here
-    produce_all_asts(&curr_dir)?;
+    produce_all_asts::<Kimchi>(&curr_dir)?;
```

Comment:
I think this is the right approach, we can have another PR later to introduce different choices of backend 👍